pkg/xds/client/client.go

```diff
@@ -368,6 +400,31 @@ func (c *Client) deletedResources(ctx context.Context, typeUrl string, curr map[
 	return deletedResources, nil
 }
 
+func (c *Client) sendACK(ctx context.Context, stream discoverypb.AggregatedDiscoveryService_StreamAggregatedResourcesClient, resp *discoverypb.DiscoveryResponse, upsertedResources map[string]map[string]proto.Message) error {
```

Comment:
Please add a comment on `sendACK` on the relationship between a non-ACK request and an ACK request?

What happens when response processing fails, is a NACK sent in that case?

Reply:
yes